pdfarranger/pdfarranger.py

```diff
@@ -163,6 +165,12 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
+def malloc_trim():
+    """Release free memory from the heap."""
+    if os.name == 'posix':
```

Comment:
This may fail on posix system like BSD. It would be more portable to try and catch `AttributeError` and `OSError`. Also may be it would be better to try only once and cache the `malloc_trim` function on success.

Reply:
Sound like a good use for singleton pattern, just return a no-op function on error.